File: api/src/main/java/cn/hirrao/javaee/service/ArticleService.java
package cn.hirrao.javaee.service;

import cn.hirrao.javaee.entity.Article;
import com.baomidou.mybatisplus.core.metadata.IPage;
import org.springframework.stereotype.Service;

import java.util.List;


@Service
public interface ArticleService {
    //返回全部文章
    List<Article> findAll();

    //根据文章id查询文章
    Article findById(long id);

    //添加文章
    void addArticle(String title, String description, String image, String content);


    void deleteArticle(long id);

    void modifyArticleInfo(long id, String title, String description, String image, String content);

    IPage<Article> articlesInfo(int curPage, int size);

    IPage<Article> searchArticleByCondition(int curPage, int size, String searchCondition, String conditionValue);
}

File: api/src/main/kotlin/cn/hirrao/javaee/service/ArticleService.kt
package cn.hirrao.javaee.service

import cn.hirrao.javaee.entity.Article
import com.baomidou.mybatisplus.core.metadata.IPage
import org.springframework.stereotype.Service

@Service
interface ArticleService {
    //返回全部文章
    fun findAll(): List<Article?>?

    //根据文章id查询文章
    fun findById(id: Long): Article?

    //添加文章
    fun addArticle(title: String?, description: String?, image: String?, content: String?)


    fun deleteArticle(id: Long)

    fun modifyArticleInfo(id: Long, title: String?, description: String?, image: String?, content: String?)

    fun articlesInfo(curPage: Int, size: Int): IPage<Article?>?

    fun searchArticleByCondition(
        curPage: Int,
        size: Int,
        searchCondition: String?,
        conditionValue: String?
    ): IPage<Article?>?
}
